Add ann opportunity to open static inventory by a player

iV0rtex opened the pull request.

iV0rtex committed:
Add ann opportunity to open static inventory by a player

etelyatn commented on Source/C_M_PT_02/PlayerInventory.h:
```diff
@@ -0,0 +1,30 @@
+ * 
+ */
+UCLASS()
+class C_M_PT_02_API UPlayerInventory final : public UUserWidget
```
Инвентарь должен быть компонентом (UActorComponent), инстанс которого необходимо будет создать в персонаже.

Также смотрю создаешь классы финальными? А вдруг мы захотим унаследовать? Если не уверен что действительно класс должен быть финальным и закрыт от наследования - лучше этого не делать.

etelyatn commented on Source/C_M_PT_02/InventoryItem.h:
```diff
@@ -0,0 +1,31 @@
+#include "InventoryItem.generated.h"
+
+UCLASS()
+class C_M_PT_02_API AInventoryItem final : public AActor
```
Эктор содержит в себе избыточный функционал, который может быть на сцене.
В нашем же случае это некий визуальный элемент, который будет отобраться в UI игрока.
Потому лучше использовать struct, UObject или UserWidget для этих целей.
Но в целом твой вариант вполне рабочий.

etelyatn reviewed (changes requested):
Хорошая работа, нужны небольшие изменения.

iV0rtex committed:
Player inventory as a player component. Inventory size can by changed via blueprints.

iV0rtex committed:
Add changes for gitignore

iV0rtex committed:
Add changes to gitignore

iV0rtex committed:
1) Add playerInventory as component via C++
2) Rename HUD_Inventory
3) Add test logic of putting item to the player inventory
4) Add Uniform Grid Panel to inventory
5) Add update logic to WBP_PlayerInventoryV4

etelyatn reviewed (approved):
Отличная работа! Принято.
